### PR TITLE
[RLlib; RELEASE BLOCKER] Fix RLlib CLI: e.g. `rllib example run cartpole-ppo` raises an error.

### PR DESCRIPTION
**Pull request by sven1977**
Signed-off-by: sven1977 <svenmika1977@gmail.com>

Fix RLlib CLI: e.g. `rllib example run cartpole-ppo` raises an error.

The fix is to give a proper file extension to temporary RLlib example config files downloaded from our github repo (instead of leaving these temp files w/o any extension). This way, our CLI code can properly detect whether the downloaded file is py or yaml.

Added a test case for this scenario.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/tuned_examples/simple_q/cartpole-simpleq-test.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ cartpole-simpleq-test:
     env: CartPole-v1
     run: SimpleQ
     stop:
-        episode_reward_mean: 150
-        timesteps_total: 50000
+        episode_reward_mean: 50.0
```
Comment (avnishn) [inline rllib/tuned_examples/simple_q/cartpole-simpleq-test.yaml:5]: is there a reason to be changing this and lowering the threshold so much? seems sus

Comment (avnishn) [reply rllib/tuned_examples/simple_q/cartpole-simpleq-test.yaml:5]: also why are you lowering the training timestep budget?

**Review by avnishn — COMMENTED** [review]
some q's :)

**Comment by xwjiang2010** [comment]
Cherry-picked from https://github.com/ray-project/ray/pull/30664

File: rllib/tuned_examples/simple_q/cartpole-simpleq-test.yaml (hunk 1 above)
Comment (sven1977) [reply rllib/tuned_examples/simple_q/cartpole-simpleq-test.yaml:5]: Oh, good question. This is just a "test" config file. So the purpose of using this is really just to demo the CLI functionality. If you would like to really learn CartPole with SimpleQ, you'd be using the "normal" `cartpole-simpleq.yaml` file (w/o the "test").

**Comment by xwjiang2010** [comment]
Hey @sven1977 , do you know why the original [PR](https://github.com/ray-project/ray/pull/30664/files) has 16 files while this one only contains 5? 

Particularly, I am looking for changes in `doc/source/rllib/rllib-algorithms.rst` - apparently release branch is failing some lint error right now: 

We are referencing to master but the master link is changed by https://github.com/ray-project/ray/pull/30664/files. This is not reflected in release branch. https://buildkite.com/ray-project/oss-ci-build-branch/builds/1285#0184b9ea-e667-4718-a665-74438016f546

**Comment by sven1977** [comment]
Checking rn ... I think it was to derisk the cherry-pick PR.

**Comment by xwjiang2010** [comment]
`tutorial_rllib` is timing out. Is this expected? 
Also https://github.com/ray-project/ray/issues/30713. We can either fix it here or in a separate PR.

**Comment by xwjiang2010** [comment]
replaced by https://github.com/ray-project/ray/pull/30746

**Pull request closed without merging**